src/cif.cc

```diff
@@ -96,24 +167,117 @@ structure_from_doc(const gemmi::cif::Document &doc,
     return structure;
 }
 
+
+
 freesasa_structure *
 freesasa_structure_from_cif(std::FILE *input,
                             const freesasa_classifier *classifier,
                             int structure_options)
 {
     const auto doc = gemmi::cif::read_cstream(input, 8192, "cif-input");
     const auto models = get_models(doc);
-    freesasa_structure *structure;
 
     if (structure_options & FREESASA_JOIN_MODELS) {
-        structure = structure_from_doc(doc, *models, classifier, structure_options);
+        return structure_from_doc(doc, *models, classifier, structure_options);
     } else {
         auto firstModel = models->begin();
         auto singleModel = std::set<int>();
         singleModel.insert(*firstModel);
 
-        structure = structure_from_doc(doc, singleModel, classifier, structure_options);
+        return structure_from_doc(doc, singleModel, classifier, structure_options);
     }
+}
 
-    return structure;
+
+
+freesasa_structure*
+freesasa_structure_from_chain(const gemmi::cif::Document doc, 
+                              const std::string &chain_name,
+                              const freesasa_classifier *classifier,
+                              int structure_options)
+{
+    std::cout << "Creating a freesasa structure from chain!" << std::endl;
+
+    return structure_from_doc(doc, chain_name, 1, classifier, structure_options);
+}
+
+
+freesasa_structure*
+freesasa_structure_from_model(const gemmi::cif::Document &doc,
+                         const std::string &model, 
+                         const freesasa_classifier *classifier,
+                         int structure_options)
+{
+    std::cout << "Creating a freesasa structure from model!" << std::endl;
+
+    return structure_from_doc(doc, model, 11, classifier, structure_options);
+}
+
+
+
+std::vector<freesasa_structure*>
+freesasa_cif_structure_array(std::FILE *input,
+                         int *n,
+                         const freesasa_classifier *classifier,
+                         int options)
+{
+    int n_models = 0, n_chains = 0;
+
+    std::vector<freesasa_structure*> ss;
+
+    const auto doc = gemmi::cif::read_cstream(input, 8192, "cif-input");
+
+    gemmi::Structure gemmi_struct = gemmi::make_structure_from_block(doc.blocks[0]);
+
+    const auto models = gemmi_struct.models;
+
+    n_models = models.size();
+
+    std::cout << "Number of models: " << n_models << std::endl;
+
+    /* only keep first model if option not provided */
+    if (!(options & FREESASA_SEPARATE_MODELS)) n_models = 1;
+
+    /* for each model read chains if requested */
+    if (options & FREESASA_SEPARATE_CHAINS) 
+    {
+        for (auto& model : models) 
```

Comment:
Am I missing something, or will this give the wrong result if we have `FREESASA_SEPARATE_MODELS`?